hypersync-client/src/lib.rs

```diff
@@ -692,6 +694,212 @@ impl Client {
     }
 }
 
+const INITIAL_RECONNECT_DELAY: std::time::Duration = std::time::Duration::from_millis(200);
+const MAX_RECONNECT_DELAY: std::time::Duration = std::time::Duration::from_secs(30);
+const MAX_CONNECTION_AGE: std::time::Duration = std::time::Duration::from_secs(24 * 60 * 60);
+/// Timeout for detecting dead connections. Server sends keepalive pings every 5s,
+/// so we timeout after 15s (3x the ping interval).
+const READ_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(15);
+
+/// Events emitted by the height stream.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum HeightStreamEvent {
+    /// Successfully connected or reconnected to the SSE stream.
+    Connected,
+    /// Received a height update from the server.
+    Height(u64),
+    /// Connection lost, will attempt to reconnect after the specified delay.
+    Reconnecting {
+        /// Duration to wait before attempting reconnection.
+        delay: Duration,
+    },
+}
+
+impl Client {
+    /// Streams archive height updates from the server via Server-Sent Events.
+    ///
+    /// Establishes a long-lived SSE connection to `/height/sse` that automatically reconnects
+    /// on disconnection with exponential backoff (200ms → 400ms → ... → max 30s).
+    ///
+    /// The stream emits [`HeightStreamEvent`] to notify consumers of connection state changes
+    /// and height updates. This allows applications to display connection status to users.
+    ///
+    /// # Returns
+    /// Channel receiver yielding [`HeightStreamEvent`]s. The background task handles connection
+    /// lifecycle and sends events through this channel.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # use std::sync::Arc;
+    /// # use hypersync_client::{Client, ClientConfig, HeightStreamEvent};
+    /// # async fn example() -> anyhow::Result<()> {
+    /// let client = Arc::new(Client::new(ClientConfig {
+    ///     url: Some("https://eth.hypersync.xyz".parse()?),
+    ///     ..Default::default()
+    /// })?);
+    ///
+    /// let mut rx = client.stream_height().await?;
+    ///
+    /// while let Some(event) = rx.recv().await {
+    ///     match event {
+    ///         HeightStreamEvent::Connected => println!("Connected to stream"),
+    ///         HeightStreamEvent::Height(h) => println!("Height: {}", h),
+    ///         HeightStreamEvent::Reconnecting { delay } => {
+    ///             println!("Reconnecting in {:?}...", delay)
+    ///         }
+    ///     }
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn stream_height(self: Arc<Self>) -> Result<mpsc::Receiver<HeightStreamEvent>> {
```

Comment:
maybe we just make this a watch channel? Since that's what's actually coming from the events. Not guaranteeing intermediary updates